lab_1/main.py

```diff
@@ -5,124 +5,153 @@
-    word = 'happy'
-    left_n = 2
-    right_n = 3
-    --> [['man', 'is'], ['dog, 'cat']]
-    """
-    pass
+    global i
+    k = 0
+    for i in range(len(tokens)):  # проверка строки на корректность
+        if 48 <= ord(' '.join(tokens)[i]) <= 57 or (128 <= ord(' '.join(tokens)[i]) <= 175 or 224 <= ord(' '.join(tokens)[i]) <= 243):
+            k += 1
+            break
+    m = 0
+    for i in range(len(word)):  # проверка слова на корректность
+        if 48 <= ord(word[i]) <= 57 or (128 <= ord(word[i]) <= 175 or 224 <= ord(word[i]) <= 243):
+            m += 1
+            break
+
+    adjacent_words = []
+    if k != 0 or m != 0 or (left_n == 0 and right_n == 0) or (i - left_n < 0 and i + 1 + right_n > len(tokens) - 1):  # если некорректные токены
+        return []
+    else:
+        for i in range(len(tokens)):
+            if word1 == tokens[i]:
+                if i - left_n > 0 and i + 1 + right_n < len(tokens) - 1:  # если не выходим за границы
+                    adjacent_words.append([tokens[i - left_n], tokens[i + right_n]])
+                elif i - left_context_size < 0:
+                    adjacent_words.append(tokens[i + right_n])
+                elif i + 1 + right_context_size > len(tokens) - 1:
+                    adjacent_words.append(tokens[i - left_n])
+    return adjacent_words
 
 
 def read_from_file(path_to_file: str) -> str:
-    """
-    Opens the file and reads its content
-    :return: the initial text in string format
-    """
-    with open(path_to_file, 'r', encoding='utf-8') as fs:
-        data = fs.read()
+    with open(path_to_file, 'r', encoding = 'utf-8') as report:
+        data = report.read()
 
     return data
 
 
 def write_to_file(path_to_file: str, content: list):
-    """
-    Writes the result in a file
-    """
-    pass
-
-
-def sort_concordance(tokens: list, word: str, left_context_size: int, right_context_size: int, left_sort: bool) -> list:
-    """
-    Gets a concordance of a word and sorts it by either left or right context
-    :param tokens: a list of tokens
-    :param word: a word-base for a concordance
-    :param left_context_size: the number of words in the left context
-    :param right_context_size: the number of words in the right context
-    :param left_sort: if True, sort by the left context, False – by the right context
-    :return: a concordance
-    e.g. tokens = ['the', 'weather', 'is', 'sunny', 'the', 'man', 'is', 'happy',
-                    'the', 'dog', 'is', 'happy', 'but', 'the', 'cat', 'is', 'sad']
-    word = 'happy'
-    left_context_size = 2
-    right_context_size = 3
-    left_sort = True
-    --> [['dog', 'is', 'happy', 'but', 'the', 'cat'], ['man', 'is', 'happy', 'the', 'dog', 'is']]
-    """
-    pass
+    with open(path_to_file, 'r') as report:
+        for i in range(len(concordance)):
+            report.write(' '.join(concordance[i]))
+            report.write('\n')
```

Comment:
it's not good to write to file at each iteration
first, write "i" to some variable 
then OUT OF "for", write this variable into a file